File: Sorting Algorithms/Selection Sort/Python/selectionsort.py
def selectionSort(arr):
    for i in range(0, len(arr)):
        minIndex = i;
        for j in range(i, len(arr)):
            if(arr[j] < arr[minIndex]):
                minIndex = j
            temp = arr[minIndex]
            arr[minIndex] = arr[i]
            arr[i] = temp
    return arr

arr = [4,5,7,2,4,7,1,9,7]
print("Array before getting sorted: ")
print(arr)

print("Array after getting sorted: ")
print(selectionSort(arr))

File: Sorting Algorithms/Selection Sort/Java/src/SelectionSort.java
public class SelectionSort {
    public static void main(String[] args) {

        int [] arr = {4,5,7,2,4,7,1,9,7};

        System.out.println("Array before getting sorted: ");
        printArray(arr);

        System.out.println("\nArray after getting sorted: ");
        printArray(selectionSort(arr));
    }

    private static int[] selectionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++){
            int minIndex = i;
            for (int j = i; j < arr.length; j++){
                if (arr[j] < arr[minIndex]){
                    minIndex = j;
                }
                int temp = arr[minIndex];
                arr[minIndex] = arr[i];
                arr[i] = temp;
            }
        }

        return arr;
    }

    private static void printArray(int[] arr) {
        for (int i = 0; i <= arr.length - 1; i++){
            System.out.print(arr[i] + " ");
        }
    }
}
